File: LeetCode/Easy/1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();

        int aLen = word1.length();
        int bLen = word2.length();

        int i = 0, j = 0;
        while(i < aLen || j < bLen) {
            if(i < aLen) sb.append(word1.charAt(i++));
            if(j < bLen) sb.append(word2.charAt(j++));
        }


        return sb.toString();
    }
}

File: LeetCode/Easy/1768-merge-strings-alternately/1768-merge-strings-alternately.kt
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        val sb = StringBuilder()

        val iLen = word1.length
        val jLen = word2.length
        var i = 0
        var j = 0

        while(i < iLen || j < jLen) {
            if(i < iLen) sb.append(word1[i++])
            if(j < jLen) sb.append(word2[j++])
        }

        return sb.toString()
    }
}